opensearch-operator/pkg/reconcilers/rollingRestart.go

```diff
@@ -126,18 +126,16 @@ func (r *RollingRestartReconciler) Reconcile() (ctrl.Result, error) {
 
 	// Restart statefulset pod.  Order is not important so we just pick the first we find
 	for _, nodePool := range r.instance.Spec.NodePools {
-		if helpers.ContainsString(nodePool.Roles, "data") {
-			sts := &appsv1.StatefulSet{}
-			if err := r.Get(r.ctx, types.NamespacedName{
-				Name:      builders.StsName(r.instance, &nodePool),
-				Namespace: r.instance.Namespace,
-			}, sts); err != nil {
-				return ctrl.Result{}, err
-			}
-			if sts.Status.UpdateRevision != "" &&
-				sts.Status.UpdatedReplicas != pointer.Int32Deref(sts.Spec.Replicas, 1) {
-				return r.restartStatefulSetPod(sts)
-			}
+		sts := &appsv1.StatefulSet{}
```

Comment:
it will also be good for a resources change rolling restart ? cause i believe it with the same func

Reply:
The new logic applies to upgrades and restarts both.